itest/rust/src/engine_tests/native_structures_test.rs

```diff
@@ -95,52 +99,49 @@ fn native_structure_codegen() {
 }
 
 #[itest]
-fn native_structure_out_parameter() {
-    // Instantiate a TextServerExtension and then have Godot call a
-    // function which uses an 'out' pointer parameter.
-    let mut ext = TestTextServer::new_gd();
-    let result = ext
-        .clone()
-        .upcast::<TextServer>()
-        .shaped_text_get_carets(Rid::new(100), 200);
-
-    // Check that we called the virtual function.
-    let cell = ext.bind_mut().cell.take();
-    assert_eq!(cell, Some((Rid::new(100), 200)));
-
-    // Check the result dictionary (Godot made it out of our 'out'
-    // param).
+fn native_structure_parameter() {
+    let caret = CaretInfo {
+        leading_caret: Rect2::from_components(0.0, 0.0, 0.0, 0.0),
+        trailing_caret: Rect2::from_components(1.0, 1.0, 1.0, 1.0),
+        leading_direction: Direction::AUTO,
+        trailing_direction: Direction::LTR,
+    };
+
+    let ptr: *const _ = &caret;
```

Comment:
Doesn't hurt to be explicit:
```suggestion
    let ptr: *const CaretInfo = &caret;
```
But you could also use `ptr::addr_of!` which is imo the more idiomatic/explicit way?